raiden-ts/src/channels/reducer.ts

```diff
@@ -33,96 +31,103 @@ const tokens = createReducer(initialState.tokens).handle(
   (state, { payload: { token, tokenNetwork } }) => ({ ...state, [token]: tokenNetwork }),
 );
 
-// Reducers for different actions
-function channelOpenRequestReducer(
-  state: RaidenState['channels'],
-  action: channelOpen.request,
-): RaidenState['channels'] {
-  const path = [action.meta.tokenNetwork, action.meta.partner];
-  if (get(path, state)) return state; // there's already a channel with partner
-  const channel: Channel = {
-    state: ChannelState.opening,
-    own: { deposit: Zero as UInt<32> },
-    partner: { deposit: Zero as UInt<32> },
-  };
-  return set(path, channel, state);
-}
+const pendingTxs: Reducer<RaidenState['pendingTxs'], RaidenAction> = (
+  state = initialState.pendingTxs,
+  action: RaidenAction,
+): RaidenState['pendingTxs'] => {
+  // filter out non-ConfirmableActions's
+  if (!isActionOf(ConfirmableActions, action)) return state;
+  // if confirmed==undefined, add action to state
+  else if (action.payload.confirmed === undefined) return [...state, action];
+  // else (either confirmed or removed), remove from state
+  else {
+    const newState = state.filter(
+      (a) => a.type !== action.type || action.payload.txHash !== a.payload.txHash,
+    );
+    if (newState.length !== state.length) return newState;
+    return state;
+  }
+};
+
+const emptyChannelEnd: ChannelEnd = {
+  address: AddressZero as Address,
+  deposit: Zero as UInt<32>,
+  withdraw: Zero as UInt<32>,
+  locks: [],
+  balanceProof: {
+    chainId: Zero as UInt<32>,
+    tokenNetworkAddress: AddressZero as Address,
+    channelId: Zero as UInt<32>,
+    nonce: Zero as UInt<8>,
+    transferredAmount: Zero as UInt<32>,
+    lockedAmount: Zero as UInt<32>,
+    locksroot: HashZero as Hash,
+    additionalHash: HashZero as Hash,
+    signature: SignatureZero,
+  },
+};
 
-function channelOpenSuccessReducer(
-  state: RaidenState['channels'],
-  action: channelOpen.success,
-): RaidenState['channels'] {
-  const path = [action.meta.tokenNetwork, action.meta.partner];
+function channelOpenSuccessReducer(state: RaidenState, action: channelOpen.success): RaidenState {
+  const key = channelKey(action.meta);
   // ignore if older than currently set channel, or unconfirmed or removed
-  if (
-    getOr(0, [...path, 'openBlock'], state) >= action.payload.txBlock ||
-    !action.payload.confirmed
-  )
+  if ((state.channels[key]?.openBlock ?? 0) >= action.payload.txBlock || !action.payload.confirmed)
     return state;
   const channel: Channel = {
     state: ChannelState.open,
-    own: { deposit: Zero as UInt<32> },
-    partner: { deposit: Zero as UInt<32> },
     id: action.payload.id,
+    token: action.payload.token,
+    tokenNetwork: action.meta.tokenNetwork,
     settleTimeout: action.payload.settleTimeout,
     isFirstParticipant: action.payload.isFirstParticipant,
     openBlock: action.payload.txBlock,
+    own: {
+      ...emptyChannelEnd,
```

Comment:
This channel end is the same structure as the one in the python client right?

Reply:
Similar, but differ on some things. I wanted to deduplicate things, and be more certain about some things, but they're related, yes.